ast/edit/task_editor.go

```diff
@@ -0,0 +1,120 @@
+package edit
+
+import (
+	"fmt"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/ast"
+	"github.com/influxdata/flux/codes"
+)
+
+// GetOption finds and returns the init for the option's variable assignment
+func GetOption(file *ast.File, name string) (ast.Expression, error) {
+	for _, st := range file.Body {
+		if val, ok := st.(*ast.OptionStatement); ok {
+			assign := val.Assignment
+			if va, ok := assign.(*ast.VariableAssignment); ok {
+				if va.ID.Name == name {
+					if ok {
+						return va.Init, nil
+					}
+				}
+			}
+		}
+	}
+
+	return nil, &flux.Error{
+		Code: codes.Internal,
+		Msg:  "Option not found",
+	}
+}
+
+// SetOption replaces an existing option's init with the provided init or adds
+// the option if it doesn't exist. The file AST is mutated in place.
+func SetOption(file *ast.File, name string, expr ast.Expression) {
+	// check for the correct file
+	for _, st := range file.Body {
+		if val, ok := st.(*ast.OptionStatement); ok {
+			assign := val.Assignment
+			if va, ok := assign.(*ast.VariableAssignment); ok {
+				if va.ID.Name == name {
+					// replace the variable assignment's init
+					va.Init = expr
+					return
+				}
+			}
+		}
+	}
+	// option was not found. prepend new option to body
+	file.Body = append([]ast.Statement{&ast.OptionStatement{
+		Assignment: &ast.VariableAssignment{
+			ID:   &ast.Identifier{Name: name},
+			Init: expr,
+		},
+	}}, file.Body...)
+}
+
+// DeleteOption removes an option if it exists. The file AST is mutated in place.
+func DeleteOption(file *ast.File, name string) error {
+	for i, st := range file.Body {
+		if val, ok := st.(*ast.OptionStatement); ok {
+			assign := val.Assignment
+			if va, ok := assign.(*ast.VariableAssignment); ok {
+				if va.ID.Name == name {
+					file.Body = append(file.Body[:i], file.Body[i+1:]...)
+					return nil
+				}
+			}
+		}
+	}
+
+	return &flux.Error{
```

Comment:
Do we want to error if the option is not found? I think callers will not be interested to know if it existed or not previously only that it has been removed if it did. Thoughts?

Reply:
Yeah, I noticed that @jlapacik didn't do this in the typescript version and asked him about it. I thought it might be relevant information, but I guess maybe not. I'll change it.